pkg/kv/kvserver/replica_raft_overload.go

```diff
@@ -193,26 +198,117 @@ func computeExpendableOverloadedFollowers(
 	return liveOverloadedVoterCandidates, nonLive
 }
 
-type overloadedStoresMap atomic.Value // map[roachpb.StoreID]*admissionpb.IOThreshold
+type ioThresholdMap struct {
+	threshold float64 // threshold at which the score indicates pausability
+	seq       int     // bumped on creation if pausable set changed
+	m         map[roachpb.StoreID]*admissionpb.IOThreshold
+}
+
+func (osm ioThresholdMap) String() string {
+	return redact.StringWithoutMarkers(osm)
+}
 
-func (osm *overloadedStoresMap) Load() map[roachpb.StoreID]*admissionpb.IOThreshold {
-	v, _ := (*atomic.Value)(osm).Load().(map[roachpb.StoreID]*admissionpb.IOThreshold)
-	return v
+var _ redact.SafeFormatter = (*ioThresholdMap)(nil)
+
+func (osm ioThresholdMap) SafeFormat(s redact.SafePrinter, verb rune) {
+	var sl []roachpb.StoreID
+	for id := range osm.m {
+		sl = append(sl, id)
+	}
+	sort.Slice(sl, func(i, j int) bool {
+		a, _ := osm.m[sl[i]].Score()
+		b, _ := osm.m[sl[j]].Score()
+		return a < b
+	})
+	for i, id := range sl {
+		if i > 0 {
+			s.SafeString(", ")
+		}
+		s.Printf("s%d: %s", id, osm.m[id])
+	}
+	if len(sl) > 0 {
+		s.Printf(" [pausable-threshold=%.2f]", osm.threshold)
+	}
+}
+
+var _ ioThresholdMapI = (*ioThresholdMap)(nil)
+
+// Pausable implements ioThresholdMapI.
+func (osm *ioThresholdMap) Pausable(id roachpb.StoreID) bool {
```

Comment:
Nit: the pausable nomenclature is a bit odd. In principle, every replica is pausable, if we decide to do so. I understand that this comes from a desire to keep the pause decision separate from the overload determination, and trigger it at a lower threshold. Maybe `ShouldPause` or something, to make it clear that this is a policy decision and not a state as such?

Reply:
I renamed to `AbovePausableThreshold` to make explicit what this is and isn't.